internal/server/data/provideruser.go

```diff
@@ -104,9 +106,16 @@ func ListProviderUsers(tx ReadTxn, providerID uid.ID, p *SCIMParameters) ([]mode
 	query.B(table.Table())
 	query.B("INNER JOIN providers ON provider_users.provider_id = providers.id AND providers.organization_id = ?", tx.OrganizationID())
 	query.B("WHERE provider_id = ?", providerID)
-
+	if p != nil && p.Filter != nil {
+		filter, err := filterSQL(*p.Filter)
+		if err != nil {
+			return nil, fmt.Errorf("apply filter: %w", err)
+		}
+		query.B("AND (")
+		query.B(filter)
```

Comment:
To fix the warning from the querylinter, and ensure that we're escaping all the parameters, I think `filterSQL` and the functions it calls can accept a `query *querybuilder.Query`. 

That way the functions themselves can build up the query and use the second argument to `query.B` to add the untrusted input.